File: src/main/java/projetShop/entities/AvisKey.java
package projetShop.entities;

import java.io.Serializable;
import java.util.Objects;

public class AvisKey implements Serializable {
	private Client client;
	private Article article;
	
	public AvisKey() {
	}

	
	public Client getClient() {
		return client;
	}

	public void setClient(Client client) {
		this.client = client;
	}

	public Article getArticle() {
		return article;
	}

	public void setArticle(Article article) {
		this.article = article;
	}

	@Override
	public int hashCode() {
		return Objects.hash(article, client);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		AvisKey other = (AvisKey) obj;
		return Objects.equals(article, other.article) && Objects.equals(client, other.client);
	}
	
	
	
	

}

File: Formation.ProjetShop/ProjetShopAngular/src/app/models/article/avis-key.ts
import { Client } from "../client/client";
import { Article } from "./article";

export class AvisKey {
    constructor(
        private _client?: Client,
        private _article?: Article
    ) { }

    public get client(): Client | undefined {
        return this._client;
    }

    public set client(value: Client | undefined) {
        this._client = value;
    }

    public get article(): Article | undefined {
        return this._article;
    }

    public set article(value: Article | undefined) {
        this._article = value;
    }
}
